packages/open-next/src/core/routing/matcher.ts

```diff
@@ -255,8 +255,8 @@ export function handleRewrites<T extends RewriteDefinition>(
 function handleTrailingSlashRedirect(
   event: InternalEvent,
 ): false | InternalResult {
-  const url = new URL(event.url, "http://localhost");
-  const emptyBody = emptyReadableStream();
+  // When rawPath is `//domain`, `url.host` would be `domain`.
```

Comment:
```suggestion
  // When rawPath is `//domain`, `url.host` would be `domain`.
  // https://github.com/opennextjs/opennextjs-aws/issues/355
```
Adding the issue here may help understanding why we need to do that